granulate_utils/metrics/sampler.py

```diff
@@ -274,6 +237,31 @@ def _init_collectors(self):
                 SparkApplicationMetricsCollector(self._cluster_mode, self._master_address, self._logger)
             )
 
+    def _validate_manual_configuration(self) -> bool:
```

Comment:
yes but metrics subpackage has 0 tests :man_shrugging:. whole sampler needs to be put under the tests, won't fix in this PR because validation logic is straightforward.